core/src/subgraph/runner.rs

```diff
@@ -106,6 +106,57 @@ where
         self.run_inner(break_on_restart).await
     }
 
+    fn build_filter(&self) -> C::TriggerFilter {
+        let current_ptr = self.inputs.store.block_ptr();
+        let static_filters = self.inputs.static_filters
+            || self.ctx.instance().hosts().len() > ENV_VARS.static_filters_threshold;
+
+        // if static_filters is enabled, build a minimal filter with the static data sources and
+        // add the necessary filters based on templates.
+        // if not enabled we just stick to the filter based on all the data sources.
+        // This specifically removes dynamic data sources based filters because these can be derived
+        // from templates AND this reduces the cost of egress traffic by making the payloads smaller.
+        let filter = if static_filters {
```

Comment:
you can get rid of this nesting, either `if static_filters { ... return filter}` and remove the else or just return the result of the if expression.

Reply:
This doesn't seem to be addressed